mm/2s2h/Enhancements/Equipment/ArrowCycle.cpp

```diff
@@ -0,0 +1,286 @@
+                           DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_LEFT) <= ITEM_BOW_LIGHT),
+                      &play->interfaceCtx.shipInterface.dpad.dLeftAlpha);
+
+    UpdateButtonAlpha(flashAlpha,
+                      (DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_DOWN) == ITEM_BOW) ||
+                          (DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_DOWN) >= ITEM_BOW_FIRE &&
+                           DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_DOWN) <= ITEM_BOW_LIGHT),
+                      &play->interfaceCtx.shipInterface.dpad.dDownAlpha);
+
+    UpdateButtonAlpha(flashAlpha,
+                      (DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_UP) == ITEM_BOW) ||
+                          (DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_UP) >= ITEM_BOW_FIRE &&
+                           DPAD_GET_CUR_FORM_BTN_ITEM(EQUIP_SLOT_D_UP) <= ITEM_BOW_LIGHT),
+                      &play->interfaceCtx.shipInterface.dpad.dUpAlpha);
+}
+
+static void UpdateEquippedBow(PlayState* play, s8 arrowType) {
+    s32 bowItem = GetBowItemForArrow(static_cast<PlayerItemAction>(arrowType));
+
+    // Update C-buttons
+    for (s32 i = 0; i < 3; i++) {
+        if ((BUTTON_ITEM_EQUIP(0, i) == ITEM_BOW) ||
+            (BUTTON_ITEM_EQUIP(0, i) >= ITEM_BOW_FIRE && BUTTON_ITEM_EQUIP(0, i) <= ITEM_BOW_LIGHT)) {
+            BUTTON_ITEM_EQUIP(0, i) = bowItem;
+            C_SLOT_EQUIP(0, i) = SLOT_BOW;
+            Interface_LoadItemIcon(play, i);
+            gSaveContext.buttonStatus[i] = BTN_ENABLED;
+            sButtonFlashTimer = BUTTON_FLASH_DURATION;
+            sButtonFlashCount = 0;
+        }
+    }
+
+    // Update D-pad
+    for (s32 i = 0; i < 4; i++) {
+        if ((DPAD_BUTTON_ITEM_EQUIP(0, i) == ITEM_BOW) ||
+            (DPAD_BUTTON_ITEM_EQUIP(0, i) >= ITEM_BOW_FIRE && DPAD_BUTTON_ITEM_EQUIP(0, i) <= ITEM_BOW_LIGHT)) {
+            DPAD_BUTTON_ITEM_EQUIP(0, i) = bowItem;
+            DPAD_SLOT_EQUIP(0, i) = SLOT_BOW;
+            Interface_Dpad_LoadItemIcon(play, i);
+            gSaveContext.shipSaveContext.dpad.status[i] = BTN_ENABLED;
+            sButtonFlashTimer = BUTTON_FLASH_DURATION;
+            sButtonFlashCount = 0;
+        }
+    }
+}
+
+// Core Arrow Cycling Function
+static void CycleToNextArrow(PlayState* play, Player* player) {
+    s8 nextArrow = GetNextArrowType(player->heldItemAction);
+
+    if (player->heldActor != NULL) {
+        EnArrow* arrow = (EnArrow*)player->heldActor;
+
+        if (arrow->actor.child != NULL) {
+            Actor_Kill(arrow->actor.child);
+        }
+
+        Actor_Kill(&arrow->actor);
+    }
+
+    Player_InitItemAction(play, player, static_cast<PlayerItemAction>(nextArrow));
+    UpdateEquippedBow(play, nextArrow);
+    Audio_PlaySfx(NA_SE_PL_CHANGE_ARMS);
+}
+
+// Registration and Hooks
+void RegisterArrowCycle() {
+    COND_VB_SHOULD(VB_MINIMAP_TOGGLE, CVAR, {
+        if (CVAR && gPlayState != NULL && CanCycleArrows()) {
```

Comment:
From what I'm reading, gPlayState should never be null in this VB handler, and it's conditionally registered based on the cvar, so you can change this to just
```suggestion
        if (CanCycleArrows()) {
```